Mlem/Views/Shared/Cached Image.swift

```diff
@@ -112,15 +126,30 @@ struct CachedImage: View {
                             shouldRecomputeSize = false
                         }
                     }
-                if shouldExpand {
-                    imageView
-                        .onTapGesture {
-                            imageDetailSheetState.url = url // show image detail
-                            onTapCallback?()
-                        }
-                } else {
-                    imageView
-                }
+                    .if(shouldExpand) { content in
+                        content
+                            .onTapGesture {
+                                imageDetailSheetState.url = url // show image detail
+                                onTapCallback?()
+                            }
+                    }
+                    .if(hasContextMenu && fixedSize == nil) { content in
```

Comment:
Propose combining the two context menu view modifiers, and moving the preview logic into a view builder function for clarity/readability:
```swift
  .if(hasContextMenu) { content in
                        content
                            .contextMenu {
                                contextMenuActions(image: image)
                            } preview: {
                                contextMenuPreview(image: image)
                            }
                    }
``` 
and
```swift
@ViewBuilder
    private func contextMenuPreview(image: Image) -> some View {
        if fixedSize != nil {
            image
                .resizable()
                .onTapGesture {
                    imageDetailSheetState.url = url
                    onTapCallback?()
                }
                .transition(.identity)
        } else {
            EmptyView()
        }
    }
```